src/Features/LanguageServer/ProtocolUnitTests/DocumentChanges/DocumentChangesTests.WithFindAllReferences.cs

```diff
@@ -75,20 +75,13 @@ await using (testLspServer)
                 Assert.Equal("M", findResults[1].ContainingMember);
                 Assert.Equal("M2", findResults[3].ContainingMember);
 
-                {
-                    // NOTE: This is not a real world scenario.
-                    //
-                    // By closing the document we revert back to the original state, but in the real world the original
-                    // state will have been updated by back channels (text buffer sync, file changed on disk, etc.) This
-                    // is validating that the above didn't succeed by any means except the FAR handler being passed the
-                    // updated document, so if we regress and get lucky, we still know about it.
-                    await DidClose(testLspServer, locationTyped.Uri);
-
-                    // In the mutating scenario, simulate this by rolling back to the text we had when the workspace was
-                    // created.
-                    if (mutatingLspWorkspace)
-                        ((ILspWorkspace)testLspServer.TestWorkspace).UpdateTextIfPresent(originalDocument.Id, await originalDocument.GetTextAsync());
-                }
```

Comment:
removed the hack here jason pointed out.  The workspaces themselves can determine what closing means wrt to their document contents.